packages/web3-eth/src/web3_subscriptions.ts

```diff
@@ -107,8 +108,43 @@ export class SyncingSubscription extends Web3Subscription<
 		return ['syncing'] as ['syncing'];
 	}
 
-	protected _processSubscriptionResult(data: SyncOutput) {
-		this.emit('data', format(syncSchema, data, super.returnFormat));
+	protected _processSubscriptionResult(
+		data:
+			| {
+					syncing: boolean;
+					status: { [key: string]: Numbers };
+			  }
+			| boolean,
+	) {
+		if (typeof data === 'boolean') {
+			this.emit('changed', data);
+		} else {
+			const mappedData: SyncOutput = {
+				startingBlock:
+					'StartingBlock' in data.status
+						? data.status.StartingBlock
+						: data.status.startingBlock,
+				currentBlock:
+					'CurrentBlock' in data.status
+						? data.status.CurrentBlock
+						: data.status.currentBlock,
+				highestBlock:
+					'HighestBlock' in data.status
+						? data.status.HighestBlock
+						: data.status.highestBlock,
+				knownStates:
+					'KnownStates' in data.status
+						? data.status.KnownStates
+						: data.status?.knownStates,
+				pulledStates:
+					'PulledStates' in data.status
+						? data.status.PulledStates
+						: data.status?.pulledStates,
+			};
```

Comment:
This is good. However, maybe we can write it as the following suggestion and so it will be dynamic with no need to update if a new property has been added later:
```suggestion
            // Uncapitylize the first letter in the properties' names (for example: data.status.StartingBlock becomes data.status.startingBlock)
			const mappedData: SyncOutput = Object.fromEntries(Object.entries(data.status).map(([ key, value ]) => [key.charAt(0).toLowerCase()+key.substring(1), value]))
```
The above code was inspired by: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries#object_transformations
You can try it here: https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCmBbADjAvDA3gKBnmAylAIYBOUAlmAOYBCANiMANYBcMADADS74DCAV1Kl4YKAyZtOPfDAASFagAt40CS3bdeeANJgQAdzBFiCCJpn4ACgPr14AExNmL2AL7ZQkEPYB0jagAKbRgAeQAjACt4YChfADNSEEQAUTFSClVAiOjY31EoDKyEFABKX0RiZEDAgG0YZngATy4YADdiegF4GABdUvQAPhhaxqbfYCUyAEEoQI5yqBAAGUN4Uj5iCHhA0oBqMd8IAXDoDJpAgEZS1o6u+H7SkNKgA

You may like to put this in a helper function and call it here if you think this would be needed later...

Thanks,

Reply:
Good idea. Thx. fixed

Reply:
Thanks Oleksii,